### ISSUE TITLE
Feature Request: Ability to Change PR Descriptions

### ISSUE THREAD

**Issue by fbartho** [danger/danger-js#700]
It would be great if there was an "easy" api for danger-js validations to adjust content in a PR description.

Specific ideas this would enable:

- Force formatting to the PR's description
- Update description with one or more branch-specific CI/shields.io image indicating the status of the branch
- Update the PR with direct download artifact links as builds succeed.

**Comment by orta** [comment]
Yeah, this is totally something Danger can do- `danger.github.utils` could include that function, it already has a few useful funcs like this

**Comment by fbartho** [comment]
@orta Thanks for the pointer. Reading GitHubUtils does indeed look like a likely place.

I just noticed this: https://github.com/danger/danger-js/blob/149713d64511c33706e929925f6e0b307593b390/source/platforms/github/GitHubUtils.ts#L119-L168

Do you think `createOrUpdatePR` is already the right util that I could use here?

**Comment by orta** [comment]
No, this is for making new PRs completely from scratch (and potentially running it again with new data later) 

I think you really just want to have an alias for calling `danger.github.api.pullRequests.update({ ...danger.github.thisPR, body: "My new Text"})` - as that's really all to takes, perhaps you could make it do something more nuanced, like keeping it's own area under the original body and adding info down there only

**Comment by fbartho** [comment]
@orta -- wait, you say an alias, does that mean that I could already do all this with that code-snippet you presented? Then I don't need an alias at all!

**Comment by brunolemos** [comment]
For anyone coming from Google, I was getting `404 Not Found` error with the code above, but this works:

```
danger.github.api.pulls.update({
  owner: danger.github.thisPR.owner,
  repo: danger.github.thisPR.repo,
  pull_number: danger.github.thisPR.pull_number || danger.github.thisPR.number, // <-- important
  title,
  body,
});
```
